schematic_api/api/routes.py

```diff
@@ -608,8 +620,48 @@ def get_viz_tangled_tree_layers(schema_url, figure_type):
 
     return layers[0]
 
+def download_manifest(input_token, manifest_id, new_manifest_name='', as_json=None):
+    """
+    Download a manifest based on a given manifest id. 
+    Args:
+        input_token: token of asset store
+        manifest_syn_id: syn id of a manifest
+        newManifestName: new name of a manifest that gets downloaded.
+        as_json: boolean; If true, return a manifest as a json. Default to True
+    Return: 
+        file path of the downloaded manifest
+    """
+    # call config_handler()
+    config_handler()
+
+    # use Synapse Storage
+    store = SynapseStorage(input_token=input_token)
+
+    # default as_json to True
+    if as_json is None:
+        as_json=True
+
+    # try logging in to asset store
+    try:
+        syn = store.login(input_token=input_token)
+    except Exception: 
+        raise SynapseAuthenticationError
+    try: 
+        md = ManifestDownload()
+        manifest_data = ManifestDownload.download_manifest(md, syn, manifest_id, new_manifest_name)
+        #return local file path
+        manifest_local_file_path = manifest_data['path']
+    except Exception: 
+        logger.error(f"Failed to download manifest: {manifest_id}")
```

Comment:
What is your reason for logging the error and raising it? (Is it for cloudwatch logs?)

Reply:
Thanks Tom. To be honest, I am not sure when I should raise an error and when I should just log it (so I did both). I think cloudwatch log could also show the error that you raise in the code, so this is not a problem. Let me remove the part that "log" error and just keep the "raise"